cfn-resources/privatelink-endpoint-service-data-federation-online-archive/cmd/resource/resource.go

```diff
@@ -233,22 +233,23 @@ func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.P
 		return *peErr, nil
 	}
 
-	pe, response, err := client.Atlas20231115002.DataFederationApi.ListDataFederationPrivateEndpoints(
+	pe, response, err := client.AtlasSDK.DataFederationApi.ListDataFederationPrivateEndpoints(
 		ctx.Background(),
 		*currentModel.ProjectId,
 	).Execute()
 
 	if err != nil {
 		return handleError(response, err)
 	}
-	endpoints := make([]interface{}, len(pe.Results))
-	for i := range pe.Results {
+	endpoints := make([]interface{}, len(pe.GetResults()))
+	for i, e := range pe.GetResults() {
+		eID := e.GetEndpointId()
 		endpoints[i] = Model{
 			ProjectId:  currentModel.ProjectId,
 			Profile:    currentModel.Profile,
-			Comment:    pe.Results[i].Comment,
-			Type:       pe.Results[i].Type,
-			EndpointId: &pe.Results[i].EndpointId,
+			Comment:    e.Comment,
+			Type:       e.Type,
+			EndpointId: &eID,
```

Comment:
curious why this was not defined directly
```suggestion
			EndpointId: e.EndpointId,
```

Reply:
It leads to a lint error since we have to use the pointer in a loop with `&e.EndpointId`